obspy/io/nordic/core.py

```diff
@@ -213,94 +241,162 @@ def _nortoevmag(mag_type):
         return "ML"
     inv_mag_mapping = {item: key for key, item in mag_mapping.items()}
     try:
-        mag = inv_mag_mapping[mag_type.upper()]
+        mag = inv_mag_mapping[mag_type]
     except KeyError:
         warnings.warn(mag_type + ' is not convertible')
         return ''
     return mag
 
 
-def readheader(sfile):
+def xyz_to_confidence_ellipsoid(errors):
```

Comment:
Maybe this should be moved to `core.event.base`, so that it has more visibility and other people can reuse it? It could even be a staticmethod on `ConfidenceEllipsoid` to create a new such object?